updates made to entities

yourgirlcodes opened the pull request:
Updates made to  Groups, Relationships, Customers- activities and entity
activity indicators files moved in order to keep with the folder structure 'style'

Commit:
#00000_ CR fixes

Commit:
#00000_ remove ts-ignore

Commit:
Merge branch 'master' of https://github.com/iBaryo/gigya-cdp-sdk

Commit:
Merge branch 'master' of https://github.com/iBaryo/gigya-cdp-sdk

Commit:
Merge branch 'master' of https://github.com/iBaryo/gigya-cdp-sdk

Commit:
adding inbound purposes

Commit:
Merge branch 'master' of https://github.com/iBaryo/gigya-cdp-sdk

Commit:
#00000_ added id to Customers entity

Commit:
Revert "#00000_ added id to Customers entity"

This reverts commit f6125b91fc72ae9246562e245ea6b5461c190f5d.

Commit:
Merge branch 'master' of https://github.com/iBaryo/gigya-cdp-sdk

Commit:
reuse compliance

Commit:
registered and added reuse compliance badge

Commit:
#00000_update merge rules entity

Commit:
#00000_update merge rules entity

Commit:
Merge branch 'master' of https://github.com/iBaryo/gigya-cdp-sdk

Commit:
updates for condition api change

Commit:
Merge branch 'master' of https://github.com/iBaryo/gigya-cdp-sdk

Commit:
change in CDP.ts for US5 datacenter

Commit:
change in CDP.ts for US5 datacenter

Commit:
Merge branch 'master' of https://github.com/iBaryo/gigya-cdp-sdk

Commit:
edit for export of CloudStorageApp + bandaid for typescript error

Commit:
remove TODO

Commit:
remove TODO and merge in origin master

Commit:
update for eu5

Commit:
miscellaneous change for us5 - placeholder for Rom to update

Commit:
miscellaneous change for us5 - placeholder for Rom to update

Commit:
fixes in CDP.ts

Commit:
update to Auth.ts

Commit:
000000_adding CalculatedIndicators Entity

Commit:
Merge branch 'master' of https://github.com/iBaryo/gigya-cdp-sdk

Commit:
Merge branch 'hotfix' of https://github.com/yourgirlcodes/gigya-cdp-sdk

Commit:
000000_added Activities to customer | added Relationships, CustomerEntity | added in schemaTypes and schemas endpoint for Groups

Commit:
merge conf

Commit:
merge conf

Commit:
x

Commit:
Revert "x"

This reverts commit d4dc6f31be68505016e1486b2ce0e612793510fb, reversing
changes made to b46af55ad42088f70d272dcfd5ad7d9f5bdb44e8.

Commit:
x

Commit:
Merge branch '000000_adjusting_entities'

# Conflicts:
#	CDPEntitiesApi.ts
#	entities/ActivityIndicator/index.ts
#	entities/Customer.ts
#	entities/index.ts

Commit:
remove new todo

Commit:
remove error

Commit:
remove todo

Commit:
00000_adding journeys for groups

Commit:
00000_remove todo

Commit:
00000_adding journeys for groups

Commit:
00000_adding journeys for groups

Commit:
00000_edit directories

Commit:
00000_edit directories

iBaryo commented:
@yourgirlcodes as we got more indicators now, i put them in a common directory - i see that you moved them out, why?

Commit:
move ActivityIndicator directory into Indicator directory

Commit:
Merge branch '000000_adjusting_entities' of https://github.com/yourgirlcodes/gigya-cdp-sdk

Commit:
add Purpose directory to fit folder structure

yourgirlcodes commented:
Initially, it looked better to have Indicator separated from ActivityIndicator - however, logically that is incorrect. 

see my update - I just changed the folder name to have an uppercase 'I' to match the folder layout. There is then an 'ActivityIndicator' directory too, within.
I added 'Purpose' directory to keep to the folder structure

iBaryo commented on entities/Purpose/Purpose.ts:
```diff
@@ -1,8 +1,8 @@
-import {ActivityIndicatorName} from "./indicators/ActivityIndicator";
-import {ActivityFieldName, ProfileFieldName} from "./common/Field";
-import {ActivitySchemaName} from "./Schema";
+import {Id, StaticEntity} from '../common';
```
rename this file to `index.ts` so imports could continue to be from `/Purpose`

yourgirlcodes replied:
will do, thank you

yourgirlcodes replied:
will do, thank you

iBaryo commented on entities/index.ts:
```diff
@@ -2,13 +2,13 @@ export * from './Workspace';
-export * from './indicators/ActivityIndicator';
-export * from './indicators/CalculatedIndicator';
+export * from './Indicator/ActivityIndicator';
+export * from './Indicator/CalculatedIndicator';
```
rename directory back to `indicators` - it's grouping of several actual config-entities

yourgirlcodes replied:
so when you group config-entities or anything not model/model-like then you keep to lowercase? We reserve uppercase directories for grouping of Model/Class entities?

iBaryo commented on entities/CustomerEntity/index.ts:
```diff
@@ -0,0 +1,15 @@
+
+export interface CustomerEntity extends SchemasBase, WithProtectedFields {}
+
+export interface SchemasBase extends WithId, WithSchema, WithDetails {
```
typo, `SchemaBase`?

yourgirlcodes replied:
typo - will update. thanks for noticing

Commit:
Code review fixes: directory naming, exporting from root file in directory, fixing typo

Commit:
Merge branch 'master' of https://github.com/iBaryo/gigya-cdp-sdk
